### ISSUE TITLE
Engine: Compile under Clang?

### ISSUE THREAD

**Issue by Ragora** [Ragora/Serious-Engine#5]
Pull Request #2 introduced Clang support, though I haven't personally tried it yet. We should see its state compared to GCC I've been trying to build with.

**Comment by salamanderrake** [comment]
Yeah first off I set up the clang support incorrectly, its supposed to be set by the user specifically and not in the project. Well cmake does not support it like that and that caused some other settisg cache issues. At this time I don't think there are any need for specialized clang flags for the build but that could change in the future, I just wanted to give people the choice to use clang or gcc.

**Comment by Ragora** [comment]
It's good, as long as it's set up correctly now to do that.
